smithyctl/command/component/package.go

```diff
@@ -162,15 +162,24 @@ func packageComponent(ctx context.Context, flags packageFlags, componentPath str
 		return err
 	}
 
-	var credsStore credentials.Store
+	var (
+		credsStore   credentials.Store
+		usePlainHTTP bool
+	)
+
 	if flags.registryAuthEnabled {
 		credsStore, err = creds.NewStaticStore(
-			flags.registryURL, flags.registryAuthUsername, flags.registryAuthPassword,
+			flags.registryURL,
+			flags.registryAuthUsername,
+			flags.registryAuthPassword,
 		)
 	} else {
-		credsStore, err = credentials.NewStoreFromDocker(credentials.StoreOptions{
-			DetectDefaultNativeStore: true,
-		})
+		credsStore, err = credentials.NewStoreFromDocker(
+			credentials.StoreOptions{
+				DetectDefaultNativeStore: true,
+			},
+		)
+		usePlainHTTP = true
```

Comment:
shouldn't this be set conditionally based on a flag or something? this looks like it shouldn't be used in production